### PR TITLE
Get client ID  before bootstrapping angular app. Manual Bootstrapping

### PR DESCRIPTION
**Pull request by camiriveron**
connects to #28 

Manual bootstrapping angular app so we can call api before and get the Github client id that is stored as an angular constant used to configure the authentication service.

### CODE REVIEW

File: src/js/franklin-dashboard.js [hunk 1]
```diff
@@ -39,119 +38,151 @@ angular
...
-    $stateProvider
-      .state('logged', {
-        url: '/dashboard',
-        templateUrl: 'dashboard/dashboard.html',
-        resolve: {
-          loginRequired: loginRequired
+  ]);
+
+//get github client id from franklin-api
+//TODO: should get this in dashboard backend
```
Comment (oudeismetis) [inline src/js/franklin-dashboard.js]: you won't get this from the dashboard backend. To keep things simple, you'll always just get this from API. We aren't worried about the 2nd API call right now.

**Comment by oudeismetis** [comment]
everything else looks good to me. Wait to see what Johnny finds. :+1:

File: src/js/franklin-dashboard.js [hunk 2]
```diff
@@ -29,8 +29,7 @@ import {
 }
 from './common'
 
-
-angular
+var franklinApp = angular
```
Comment (recursivefunk) [inline src/js/franklin-dashboard.js]: Lets stick to `const`

File: src/js/franklin-dashboard.js [hunk 3]
```diff
@@ -39,119 +38,151 @@ angular
...
-        url: '/detail',
-        templateUrl: 'dashboard/repo-detail.html'
-      })
-      .state('logout', {
-        url: '/login',
-        templateUrl: 'login/login.html',
-        resolve: {
-          skipIfLoggedIn: skipIfLoggedIn
+
+        function loginRequired($q, $location, $auth) {
```
Comment (recursivefunk) [inline src/js/franklin-dashboard.js]: I'd rethink this function a bit. It doesn't always resolve/reject a promise. Should probably refactor it to always resolve or reject. And let whomever calls the function decide whether or not to take you to the `/login` view.

File: src/js/franklin-dashboard.js [hunk 4]
```diff
@@ -39,119 +38,151 @@ angular
...
+          var deferred = $q.defer();
```
Comment (recursivefunk) [inline src/js/franklin-dashboard.js]: Death to `var`

**Pull request merged**